File: 2243.计算字符串的数字和.js
/*
 * @lc app=leetcode.cn id=2243 lang=javascript
 *
 * [2243] 计算字符串的数字和
 */

// @lc code=start
/**
 * @param {string} s
 * @param {number} k
 * @return {string}
 */
var digitSum = function(s, k) {
    while (s.length > k) {
        let newS = '';
        for (let i = 0; i < s.length; i += k) {
            const segment = s.slice(i, i + k);
            const sum = segment.split('').reduce((acc, curr) => acc + Number(curr), 0);
            newS += sum.toString();
        }
        s = newS;
    }
    return s;
};
// @lc code=end

File: 2243.计算字符串的数字和.java
/*
 * @lc app=leetcode.cn id=2243 lang=java
 *
 * [2243] 计算字符串的数字和
 */

// @lc code=start
class Solution {
    public String digitSum(String s, int k) {
        while (s.length() > k) {
            StringBuilder sb = new StringBuilder();
            for (int i = 0; i < s.length(); i += k) {
                int sum = 0;
                for (int j = i; j < Math.min(i + k, s.length()); j++) {
                    sum += s.charAt(j) - '0';
                }
                sb.append(sum);
            }
            s = sb.toString();
        }
        return s;
    }
}
// @lc code=end

